### PR TITLE
[6.3 Feature] SSH Key in User Templates API Stubs

### PR DESCRIPTION
**Pull request by jyejare**
$subject

### CODE REVIEW

File: tests/foreman/api/test_user.py [hunk 1]
```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
                 # pylint:disable=no-member
                 set([role.id for role in chosen_roles] + [self.anon_role.id]),
             )
+
+
+class SshKeyInUser(APITestCase):
```
Comment (oshtaier) [inline tests/foreman/api/test_user.py]: `SSHKeyInUserTestCase`

File: tests/foreman/api/test_user.py [hunk 2]
```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
...
+    @stubbed
+    @tier1
+    def test_positive_create_ssh_key_super_admin(self):
+        """SSH Key can be added to Super Admin user
+
+        :id: 397eea22-759c-4cd4-bda1-0e7835566c72
+
+        :steps:
+
+            1. Add SSH Key to the Super Admin user
```
Comment (oshtaier) [inline tests/foreman/api/test_user.py]: redundant as that step is already present in the description

File: tests/foreman/api/test_user.py [hunk 3]
```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
...
+            1. Create new user with all the details
+            2. Attempt to add invalid string as SSH Key to above user
+                e.g blabla
+
+        :expectedresults: Invalid SSH key should not be added in user
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_error_create_invalid_ssh_key(self):
```
Comment (oshtaier) [inline tests/foreman/api/test_user.py]: redundant test case. Verify that fact in the test case above

File: tests/foreman/api/test_user.py [hunk 4]
```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
...
+            2. Attempt to add invalid string as SSH Key to above user
+                e.g blabla
+
+        :expectedresults: Satellite returns 'Fingerprint could not be
+            generated' error
+        """
+
+    @stubbed
+    @tier1
+    def test_positive_error_create_invalid_length_ssh_key(self):
```
Comment (oshtaier) [inline tests/foreman/api/test_user.py]: it should be a negative test case and you should state in the description not the error as main goal of the case, but the fact itself, like:
```
Attempt to add SSH key that has invalid length
```
and in expcted result:
```
error is raised
```

File: tests/foreman/api/test_user.py [hunk 5]
```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
...
+
+            1. Create new user with all the details
+            2. Attempt to add invalid length of SSH Key to above user
+
+        :expectedresults: Satellite returns 'Length could not be calculated'
+            error
+        """
+
+    @stubbed
+    @tier3
```
Comment (oshtaier) [inline tests/foreman/api/test_user.py]: `tier1`

File: tests/foreman/api/test_user.py [hunk 6]
```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
...
+            1. Create user with all the details
+            2. Add ssh key in above user
+            3. Provision a new host from the above user
+            4. Check new hosts ENC output
+
+        :expectedresults: SSH key should be added to host ENC output
+        """
+
+    @stubbed
+    @tier2
```
Comment (oshtaier) [inline tests/foreman/api/test_user.py]: `tier1`

File: tests/foreman/api/test_user.py [hunk 7]
```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
...
+
+        :id: 4b70a950-e777-4b2d-a83d-29279715fe6d
+
+        :steps:
+
+            1. Create user with all the details
+            2. Add ssh key in above user
+            3. Provision a new host from the above user
+            4. Check new hosts ENC output
+
```
Comment (oshtaier) [inline tests/foreman/api/test_user.py]: `:CaseLevel: System`

File: tests/foreman/api/test_user.py (hunk 4 above)
Comment (svtkachenko) [inline tests/foreman/api/test_user.py]: +1 for negative.
However, as I understood from userstory, it is needed to validate several negative cases that produce different error messages, so perhaps it is better to specify that message.

File: tests/foreman/api/test_user.py [hunk 8]
```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
...
+        :id: d00905f6-3a70-4e2f-a5ae-fcac18274bb7
+
+        :steps:
+
+            1. Create new user with all the details
+            2. Add SSH key to the above user
+
+        :expectedresults: SSH key should be added to user
+        """
```
Comment (svtkachenko) [inline tests/foreman/api/test_user.py]: `:CaseLevel: Critical` for all the tier1 tests.

File: tests/foreman/api/test_user.py [hunk 9]
```diff
@@ -372,3 +372,168 @@ def test_positive_update(self):
...
+
+            1. Create user with all the details
+            2. Add SSH key in above user
+            3. List all the ssh keys of above user
+
+        :expectedresults: Satellite should list all the SSH keys of user
+        """
+
+    @stubbed
+    @tier2
```
Comment (svtkachenko) [inline tests/foreman/api/test_user.py]: Not sure that it is tier2, looks same for me as other tier1 tests.
Also `:CaseLevel: System` is missing for tier2.

Comment (oshtaier) [inline tests/foreman/api/test_user.py]: `:CaseLevel: Integration` for tier2

Comment (svtkachenko) [inline tests/foreman/api/test_user.py]: Yeah, right. Sorry :)

File: tests/foreman/api/test_user.py (hunk 5 above)
Comment (jyejare) [inline tests/foreman/api/test_user.py]: No, its not creating SSH key, but it checks if the SSH Key is associated with host enc. Updating it to tier2 test.

File: tests/foreman/api/test_user.py (hunk 7 above)
Comment (jyejare) [inline tests/foreman/api/test_user.py]: As per above comment that makes it tier2 test so cant put caselevel now

File: tests/foreman/api/test_user.py (hunk 8 above)
Comment (jyejare) [inline tests/foreman/api/test_user.py]: You mean CaseImportance ??

File: tests/foreman/api/test_user.py (hunk 9 above)
Comment (jyejare) [inline tests/foreman/api/test_user.py]: I think we dont write CaseLevel for all the tests, it is defined only once at module Level. CaseLevel has been set to Acceptance at module level.

**Comment by jyejare** [comment]
@svtkachenko @oshtaier , I have fixed your comments on PR. Please revisit for further action.

File: tests/foreman/api/test_user.py (hunk 8 above)
Comment (svtkachenko) [inline tests/foreman/api/test_user.py]: Yes, you're right.

**Comment by oshtaier** [comment]
Explained the CaseLevel idea in another PR. Please, fix current PR accordingly

**Comment by jyejare** [comment]
@oshtaier , I have just updated the CaseLevels. Please revisit and merge.

File: tests/foreman/api/test_user.py [hunk 10]
```diff
@@ -372,3 +372,167 @@ def test_positive_update(self):
...
+            2. Attempt to add invalid length of SSH Key to above user
+
+        :expectedresults: Satellite should raise 'Length could not be
+            calculated' error
+
+        :CaseImportance: Critical
+        """
+
+    @stubbed
+    @tier3
```
Comment (oshtaier) [inline tests/foreman/api/test_user.py]: doesn't sound as tier 3 test case. What system components are used here? Only user and ssh keys as in previous tier1 test cases

**Comment by oshtaier** [comment]
ACK pending last comment

**Pull request merged**